src/rateLimiters/fixedWindow.ts

```diff
@@ -0,0 +1,129 @@
+     *
+     * @param {string} uuid - unique identifer used to throttle requests
+     * @param {number} timestamp - time the request was recieved
+     * @param {number} [tokens=1]  - complexity of the query for throttling requests
+     * @return {*}  {Promise<RateLimiterResponse>}
+     * @memberof FixedWindow
+     */
+    async processRequest(
+        uuid: string,
+        timestamp: number,
+        tokens = 1
+    ): Promise<RateLimiterResponse> {
+        // set the expiry of key-value pairs in the cache to 24 hours
+        const keyExpiry = 86400000;
+
+        // attempt to get the value for the uuid from the redis cache
+        const windowJSON = await this.client.get(uuid);
+
+        if (windowJSON === null) {
+            const newUserWindow: RedisWindow = {
+                currentTokens: tokens > this.capacity ? 0 : tokens,
+                fixedWindowStart: timestamp,
+            };
+
+            if (tokens > this.capacity) {
+                await this.client.setex(uuid, keyExpiry, JSON.stringify(newUserWindow));
+                return { success: false, tokens: this.capacity };
+            }
+            await this.client.setex(uuid, keyExpiry, JSON.stringify(newUserWindow));
+            return { success: true, tokens: this.capacity - newUserWindow.currentTokens };
+        }
+        const window: RedisWindow = await JSON.parse(windowJSON);
+
+        // update the currentToken until reaches its capacity
+        window.currentTokens += tokens;
+        if (window.currentTokens > this.capacity) {
+            window.currentTokens = this.capacity;
+            return {
+                success: false,
+                tokens: this.capacity - window.currentTokens,
+            };
+        }
+
+        // update a new time window, check the current capacity situation
+        const updatedUserWindow = this.updateTimeWindow(window, timestamp);
+        if (tokens > this.capacity) {
+            await this.client.setex(uuid, keyExpiry, JSON.stringify(updatedUserWindow));
+            return { success: false, tokens: this.capacity };
+        }
+        await this.client.setex(uuid, keyExpiry, JSON.stringify(updatedUserWindow));
+        return {
+            success: true,
+            tokens: this.capacity - updatedUserWindow.currentTokens,
+        };
+    }
+
+    /**
+     * Resets the rate limiter to the intial state by clearing the redis store.
+     */
+    public reset(): void {
+        this.client.flushall();
+    }
+
+    private updateTimeWindow = (window: RedisWindow, timestamp: number): RedisWindow => {
+        const updatedUserWindow: RedisWindow = {
+            currentTokens: window.currentTokens,
+            fixedWindowStart: window.fixedWindowStart,
+        };
+        if (timestamp >= window.fixedWindowStart + this.windowSize) {
+            updatedUserWindow.fixedWindowStart = window.fixedWindowStart + this.windowSize;
```

Comment:
What if the timestamp is greater than 2 times the `windowSize` (or more, like after 1 - 2 hours). You would still only be updating the new window to start after `windowSize` when it should probably start sometime close to the `timestamp`

Reply:
this was implemented in `slidingWindowCounter`, feel free to reference

Reply:
Updated, sliding window and token bucket are always my best reference!!! Thanks, guys.